pkg/snet/network.go

```diff
@@ -104,73 +137,139 @@ func New(conf Config, eb *appevent.Broadcaster) *Network {
+		clients.StcpC.SetLogger(logging.MustGetLogger("snet.stcpC"))
 	}
 
-	return NewRaw(conf, dmsgC, stcpC)
+	if conf.NetworkConfigs.STCPR != nil {
+		ar, err := arclient.NewHTTP(conf.NetworkConfigs.STCPR.AddressResolver, conf.PubKey, conf.SecKey)
+		if err != nil {
+			return nil, err
+		}
+
+		addressResolver = ar
+
+		clients.StcprC = stcpr.NewClient(conf.PubKey, conf.SecKey, addressResolver, conf.NetworkConfigs.STCPR.LocalAddr)
+		clients.StcprC.SetLogger(logging.MustGetLogger("snet.stcprC"))
+	}
+
+	if conf.NetworkConfigs.STCPH != nil {
+		// If address resolver is not already created or if stcpr and stcph address resolvers differ
+		if conf.NetworkConfigs.STCPR == nil || conf.NetworkConfigs.STCPR.AddressResolver != conf.NetworkConfigs.STCPH.AddressResolver {
+			ar, err := arclient.NewHTTP(conf.NetworkConfigs.STCPH.AddressResolver, conf.PubKey, conf.SecKey)
+			if err != nil {
+				return nil, err
+			}
+
+			addressResolver = ar
+		}
+
+		clients.StcphC = stcph.NewClient(conf.PubKey, conf.SecKey, addressResolver)
+		clients.StcphC.SetLogger(logging.MustGetLogger("snet.stcphC"))
+	}
+
+	return NewRaw(conf, clients), nil
 }
 
 // NewRaw creates a network from a config and a dmsg client.
-func NewRaw(conf Config, dmsgC *dmsg.Client, stcpC *stcp.Client) *Network {
+func NewRaw(conf Config, clients NetworkClients) *Network {
 	networks := make([]string, 0)
 
-	if dmsgC != nil {
-		networks = append(networks, DmsgType)
+	if clients.DmsgC != nil {
+		networks = append(networks, dmsg.Type)
+	}
+
+	if clients.StcpC != nil {
+		networks = append(networks, stcp.Type)
 	}
 
-	if stcpC != nil {
-		networks = append(networks, STCPType)
+	if clients.StcprC != nil {
+		networks = append(networks, stcpr.Type)
+	}
+
+	if clients.StcphC != nil {
+		networks = append(networks, stcph.Type)
 	}
 
 	return &Network{
 		conf:     conf,
 		networks: networks,
-		dmsgC:    dmsgC,
-		stcpC:    stcpC,
+		clients:  clients,
 	}
 }
 
 // Init initiates server connections.
 func (n *Network) Init() error {
-	if n.dmsgC != nil {
+	if n.clients.DmsgC != nil {
 		time.Sleep(200 * time.Millisecond)
-		go n.dmsgC.Serve()
+		go n.clients.DmsgC.Serve()
 		time.Sleep(200 * time.Millisecond)
 	}
 
-	if n.conf.STCP != nil {
-		if n.stcpC != nil && n.conf.STCP.LocalAddr != "" {
-			if err := n.stcpC.Serve(n.conf.STCP.LocalAddr); err != nil {
+	if n.conf.NetworkConfigs.STCP != nil {
+		if n.clients.StcpC != nil && n.conf.NetworkConfigs.STCP.LocalAddr != "" {
+			if err := n.clients.StcpC.Serve(n.conf.NetworkConfigs.STCP.LocalAddr); err != nil {
 				return fmt.Errorf("failed to initiate 'stcp': %v", err)
 			}
 		} else {
 			fmt.Println("No config found for stcp")
 		}
 	}
 
+	if n.conf.NetworkConfigs.STCPR != nil {
+		if n.clients.StcprC != nil && n.conf.NetworkConfigs.STCPR.LocalAddr != "" {
+			if err := n.clients.StcprC.Serve(); err != nil {
+				return fmt.Errorf("failed to initiate 'stcpr': %w", err)
+			}
+		} else {
+			fmt.Println("No config found for stcpr")
```

Comment:
Shouldn't we use logger here?

Reply:
I agree. I copied this from `stcp` and didn't change anything. The same applies to your comments below where there's no reply from me.